inc/class-editor.php

```diff
@@ -5,8 +5,10 @@
 use WP_Post;
 
 class Editor {
+	public static Importer $importer;
+
 	public function __construct( Importer $importer ) {
-		$this->importer = $importer;
+		self::$importer = $importer;
```

Comment:
This doesn't really make sense; it's setting a static property when constructing the object itself, which means that any other instances are going to have the property overridden.

What's the root linting error that needs fixing here?

Reply:
The linting issue is the references to `$this` in static functions lower down. The functions in question need to reference this `$importer` to call it's methods.  Some of those functions are called via filters/actions.
 
I couldn't find an example of this class being used anywhere in the project so I couldn't see how it was called. It sorta feels like it wants to be a singleton as the only thing it actually needs in it's methods is this importer.  Everything else the methods use is passed in or obtained from external function calls.

Reply:
There doesn't seem to be any reason for the function `redirect_o2_edit_link_to_github` to be static so it'd be better to make that non-static, than to make this class into some sort of singleton.

Searching GitHub reveals this version on wordpress.org with the fix https://github.com/WordPress/wordpress.org/blob/master/wordpress.org/public_html/wp-content/plugins/wporg-markdown/inc/class-editor.php#L129

Reply:
@roborourke I've made those changes, thanks.
I was being extra cautious trying not to break things in case the static function is referred to elsewhere as `Editor::redirect_o2_edit_link_to_github()` But I don't know where, if anywhere, this code is used from this repo.

Reply:
iirc it was only used for the old docs.humanmade.com site that was made for Enterprise Kit so it might not be used any more.